dhis-2/dhis-services/dhis-service-field-filtering/src/main/java/org/hisp/dhis/fieldfiltering/FieldPathHelper.java

```diff
@@ -238,7 +238,12 @@ else if ( FieldPreset.SIMPLE.equals( preset.getName() ) )
             }
         }
 
-        fieldPaths.forEach( fp -> fieldPathMap.put( fp.toFullPath(), fp ) );
+        fieldPaths.forEach( fp -> {
+            if ( !fieldPathMap.containsKey( fp.toFullPath() ) )
+            {
+                fieldPathMap.put( fp.toFullPath(), fp );
```

Comment:
I guess you could use `putIfAbsent` to avoid the if and allow to write this in a lambda without the curlies.